File: DesignPatternJava/src/behavioral/mediator/ChatMediatorImpl.java
package behavioral.mediator;

import java.util.ArrayList;
import java.util.List;

/**
 * @author tamdx
 */
public class ChatMediatorImpl extends ChatMediator{
    private List<User> lstUser = new ArrayList<>();

    public ChatMediatorImpl(String groupName) {
        System.out.println("Start group chat "+groupName);
    }

    @Override
    public void sendMessage(String msg, User user) {
        lstUser.stream().filter(u-> !u.equals(user)).forEach(u->u.receive(msg));
    }

    @Override
    public void addUser(User user) {
        if(!lstUser.contains(user)){
            lstUser.add(user);
            user.setMediator(this);
        }
    }
}

File: DesignPatternPython/Behavioral/mediator/chat_mediator_impl.py
from Behavioral.mediator.chat_mediator import ChatMediator
from Behavioral.mediator.user import User
from typing import List


class ChatMediatorImp(ChatMediator):
    __lstUser: List[User] = []

    def __init__(self,groupName) -> None:
        super().__init__()
        print("%s group already created"%(groupName))

    def sendMessage(self, msg, user: User):
        [us.receive(msg) for us in list(filter(lambda u: u.name != user.name, self.__lstUser))]

    def addUser(self, user: User):
        users = list(filter(lambda u: u.name ==user.name, self.__lstUser))
        if not users:
            self.__lstUser.append(user)
            user.mediator = self